src/agents/MonarchAgent.sol

```diff
@@ -27,4 +47,57 @@ contract MonarchAgentV1 is IMonarchAgent {
 
         emit RebalancerSet(msg.sender, rebalancer);
     }
+
+    /**
+     * @notice Users can revoke the authorization of a rebalancer
+     */
+    function revoke() external {
+        delete rebalancers[msg.sender];
+
+        emit RebalancerSet(msg.sender, address(0));
+    }
+
+    /**
+     * @notice Rebalances the user's position from one market to another
+     * @param onBehalf The user to rebalance the position for
+     * @param token The token to rebalance
+     * @param fromMarkets The markets to withdraw assets from
+     * @param toMarkets The markets to supply assets to
+     */
+    function rebalance(
+        address onBehalf,
+        address token,
+        RebalanceMarketParams[] calldata fromMarkets,
+        RebalanceMarketParams[] calldata toMarkets
+    ) external payable onlyRebalancer(onBehalf) {
+        require(onBehalf == msg.sender || morphoBlue.isAuthorized(onBehalf, address(this)), ErrorsLib.AGENT_NOT_AUTHORIZED);
```

Comment:
i don't think we need this constraint: 
* `onBehalf == msg.sender` is already checked in `onlyRebalancer`
* `morphoBlue.isAuthorized` is unnecessary cuz it will be done when withdrawing from morpho (it will revert if it's not set)

Reply:
removed this constraint